File: biblioteca.java
import java.util.ArrayList;
import java.util.Objects;


public class biblioteca{

    private final ArrayList<libro>libros;

    public biblioteca() {
        libros = new ArrayList<>();
        libros.add(new libro("1984", "George Orwell", 0, true));
        libros.add(new libro("To Kill a Mockingbird", "Harper Lee", 1, true));
        libros.add(new libro("The Great Gatsby", "F. Scott Fitzgerald", 2, true));
        libros.add(new libro("Moby Dick", "Herman Melville", 3, true));
        libros.add(new libro("Pride and Prejudice", "Jane Austen", 4, true));
        libros.add(new libro("War and Peace", "Leo Tolstoy", 5, true));
        libros.add(new libro("The Odyssey", "Homer", 6, true));
        libros.add(new libro("Ulysses", "James Joyce", 7, true));
        libros.add(new libro("The Catcher in the Rye", "J.D. Salinger", 8, true));
        libros.add(new libro("The Divine Comedy", "Dante Alighieri", 9, true));
    }

    public  void agregarLibro(String titulo, String autor, int isbn ){
        libros.add(new libro(titulo, autor, isbn, true));
        System.out.println("Libro agregado: " + titulo);
    }

    public int tamLista(){
        return libros.size();
    }

    public libro buscarLibro(int isbb){
        if(libros.isEmpty()){
            return null;
        }
        for (libro libro1 : libros){
            if(Objects.equals(libro1.getIsbn(), isbb)){
                return libro1;
            }
        }
        return null;
    }

    public libro prestarLibro(int ISBN){
        for (libro libro1 : libros){
            if(Objects.equals(libro1.getIsbn(), ISBN) && libro1.getEstado() ){
                libro1.setEstado(false);
                return libro1;
            }
        }
        return null;
    }

    public libro regresarLibro(int ISBN){
        for (libro libro1 : libros){
            if(Objects.equals(libro1.getIsbn(), ISBN) && (!libro1.getEstado())) {
                libro1.setEstado(true);
                return libro1;
            }
        }
        return null;
    }
}

File: biblioteca.py
from libro import Libro

class Biblioteca:
    def __init__(self):
        self.libros = []

    def agregar_libro(self, titulo, autor, isbn):
        nuevo_libro = Libro()
        nuevo_libro.titulo = titulo
        nuevo_libro.autor = autor
        nuevo_libro.isbn = isbn
        self.libros.append(nuevo_libro)
        print(f"Libro agregado: {titulo}")

    def mostrar_libros(self):
        if not self.libros:
            print("La biblioteca está vacía.")
        else:
            print("\nLista de libros en la biblioteca:")
            for libro in self.libros:
                print(f"Título: {libro.titulo}, Autor: {libro.autor}, ISBN: {libro.isbn}, Estado: {libro.estado}")

    def buscar_libro(self, isbn):
        for libro in self.libros:
            if libro.isbn == isbn:
                return libro
        return None

    def prestar_libro(self, isbn):
        libro = self.buscar_libro(isbn)
        if libro and libro.estado == "Disponible":
            libro.estado = "Prestado"
            print(f"El libro '{libro.titulo}' ha sido prestado.")
        elif libro:
            print(f"El libro '{libro.titulo}' ya está prestado.")
        else:
            print(f"No se encontró ningún libro con ISBN {isbn}.")

    def regresar_libro(self, isbn):
        libro = self.buscar_libro(isbn)
        if libro and libro.estado == "Prestado":
            libro.estado = "Disponible"
            print(f"El libro '{libro.titulo}' ha sido regresado.")
        elif libro:
            print(f"El libro '{libro.titulo}' ya está disponible.")
        else:
            print(f"No se encontró ningún libro con ISBN {isbn}.")
        return None